src/NSubstitute.Analyzers.Shared/DiagnosticAnalyzers/AbstractNonSubstitutableMemberReceivedAnalyzer.cs

```diff
@@ -117,5 +117,12 @@ private SyntaxNode GetKnownParent(SyntaxNode receivedSyntaxNode)
 
             return null;
         }
+
+        private Location GetSymbolLocation(SyntaxNode syntaxNode, ISymbol symbol)
+        {
+            var actualNode = symbol is IMethodSymbol _ ? syntaxNode.Parent : syntaxNode;
```

Comment:
When searching for the method given Received call belongs to I always take the first syntax node (from the one I've defined as allowed). However in method invocations like
````c#
substitute.DidNotReceive().Foo2();
````
this will give you SimpleMemberAccessExpression so only `substitute.DidNotReceive().Foo2` part. In order to get whole invocation expression, I need to get Parent node as well which will give me the `substitute.DidNotReceive().Foo2()`